aiida_common_workflows/workflows/dissociation.py

```diff
@@ -0,0 +1,216 @@
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super().define(spec)
+        spec.input('molecule', valid_type=orm.StructureData, validator=validate_molecule, help='The input molecule.')
+        spec.input('distances', valid_type=orm.List, required=False, validator=validate_distances,
+            help='The list of dinstances in Ang at which the total energy of the molecule should be computed.')
+        spec.input('distances_count', valid_type=orm.Int, default=lambda: orm.Int(20),
+            validator=validate_distances_count,
+            help='The number of points to compute in the dissociation curve.')
+        spec.input('distance_min', valid_type=orm.Float, default=lambda: orm.Float(0.5),
+            validator=validate_distance_min,
+            help='The minimum tested dinstance in Ang.')
+        spec.input('distance_max', valid_type=orm.Float, default=lambda: orm.Float(3),
+            validator=validate_distance_max,
+            help='The maximum tested distance in Ang.')
+        spec.input_namespace('generator_inputs', dynamic=True,
+            help='The inputs that will be passed to the inputs generator of the specified `sub_process`.')
+        spec.input('generator_inputs.calc_engines', valid_type=dict, non_db=True)
+        spec.input('generator_inputs.protocol', valid_type=str, non_db=True,
+            help='The protocol to use when determining the workchain inputs.')
+        spec.input('generator_inputs.relax_type', valid_type=RelaxType, non_db=True, validator=validate_relax,
+            help='The type of relaxation to perform.')
+        spec.input('generator_inputs.spin_type', valid_type=SpinType, required=False, non_db=True,
+            help='The type of spin for the calculation.')
+        spec.input('generator_inputs.electronic_type', valid_type=ElectronicType, required=False, non_db=True,
+            help='The type of electronics (insulator/metal) for the calculation.')
+        spec.input('generator_inputs.magnetization_per_site', valid_type=orm.List, required=False, non_db=True,
+            help='List containing the initial magnetization fer each site.')
+        spec.input('generator_inputs.threshold_forces', valid_type=float, required=False, non_db=True,
+            help='Target threshold for the forces in eV/Å.')
+        spec.input('generator_inputs.threshold_stress', valid_type=float, required=False, non_db=True,
+            help='Target threshold for the stress in eV/Å^3.')
+        spec.input_namespace('sub_process', dynamic=True, populate_defaults=False)
+        spec.input('sub_process_class', non_db=True, validator=validate_sub_process_class)
+        spec.inputs.validator = validate_inputs
+        spec.outline(
+            cls.run_init,
+            cls.run_dissociation,
+            cls.inspect_results,
+        )
+        spec.output('curve_data', valid_type=orm.ArrayData,
+            help='The computed total energy (and eventually spin) versus distance.')
```

Comment:
Why don't we keep the same idea as for the EOS workchain and do something like the following:
```
spec.output_namespace('distances', valid_type=Float)
spec.output_namespace('total_energies', valid_type=Float)
```
In this way, there is no need to create new output nodes, you are simply returning existing inputs/outputs

Reply:
I was just worried that it would return too many outputs. To have a decent curve at least 20 distances are needed. Isn't it a concern the amount of outputs?